specification/app/data-plane/Microsoft.App.DynamicSessions/preview/2024-10-02-preview/DynamicSessions.json

```
@@ -0,0 +1,793 @@
+        },
+        "executionTimeInMilliseconds": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The execution time of the code in milliseconds."
+        }
+      }
+    },
+    "SessionResourceFile": {
+      "type": "object",
+      "description": "Code execution file resource.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the file.",
+          "x-ms-mutability": [
+            "read",
+            "update",
+            "create"
+          ]
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the session resource file."
+        },
+        "contentType": {
+          "type": "string",
+          "description": "The type of the content of this file."
+        },
+        "sizeInBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The size of the file."
+        },
+        "lastModifiedAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The date time in RFC3339 format when the file was last modified."
+        }
+      },
+      "required": [
+        "name",
+        "type",
+        "lastModifiedAt"
+      ]
+    },
+    "SessionResourceFileCollection": {
+      "type": "object",
+      "description": "Code execution file resource collection.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The files uploaded.",
+          "items": {
+            "$ref": "#/definitions/SessionResourceFile"
+          },
+          "x-ms-identifiers": []
+        }
+      },
+      "required": [
+        "value"
+      ]
+    }
+  },
+  "parameters": {
+    "Azure.Core.Foundations.ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "description": "The API version to use for this operation.",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "x-ms-parameter-location": "method",
+      "x-ms-client-name": "apiVersion"
+    },
+    "CodeExecutionIdentifier": {
+      "name": "executionId",
+      "in": "path",
+      "description": "The identifier of code execution.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ExecutionOperationIdHeader": {
+      "name": "operation-id",
+      "in": "header",
+      "description": "The id of this execution operation.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "x-ms-client-name": "operationId"
+    },
+    "SessionIdentifier": {
+      "name": "identifier",
+      "in": "query",
+      "description": "The identifier of the session.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```

My recollection from a discussion over email is that this parameter is user-specified, so a) that should be stated in the description, and b) there should be min/max length and pattern constraints specified (similar to a path parameter). I included example values in this suggestion but you should change these to match the service behavior.
```suggestion
    "SessionIdentifier": {
      "name": "identifier",
      "in": "query",
      "description": "The user-assigned identifier of the session.",
      "required": true,
      "type": "string",
      "minLength": 1,
      "maxLength": 64,
      "pattern": "^[A-Za-z0-9-_]*$",
      "x-ms-parameter-location": "method"
    },
```

---

Fixed